specs/ExecuteScriptWithResult.md

```diff
@@ -0,0 +1,315 @@
+            // if the result type isn't string type.
+            try {
+                String stringResult = result.TryGetResultAsString();
+                Debug.WriteLine($"replaced string: {result.stringResult}");
+            }
+            catch (ArgumentException) {
+                Debug.WriteLine($"Non-string message received");
+            }
+        }
+        // Here is the case of execution exception.
+        else
+        {
+            var exception = result.Exception;
+            String exceptionInfo = "The script execution failed." + 
+                "\nName:" + exception.Name +
+                "\nMesssge: " + exception.Message +
+                "\nLineNumber:" + exception.LineNumber + 
+                ", ColumnNumber:" + exception.ColumnNumber;
+            Debug.WriteLine($"{exceptionInfo}");
+        }
+    }
+}
+```
+
+# API Details
+## Win32 C++
+```c++
+/// This interface represents a JavaScript exception.
+/// If the CoreWebView2.ExecuteScriptWithResult result has IsSuccessful as false,
+/// you can use the result's Exception property to get the script exception.
+[uuid(82F22B72-1B22-403E-A0B9-A8816C9C8E45), object, pointer_default(unique)]
+interface ICoreWebView2ExecuteScriptException : IUnknown {
+
+  /// The line number of the source where the exception occurred.
+  /// In the JSON it is `exceptionDetail.lineNumber`.
+  /// Note that this position starts at 0.
+  [propget] HRESULT LineNumber([out, retval] UINT32* value);
+  
+  /// The column number of the source where the exception occurred.
+  /// In the JSON it is `exceptionDetail.columnNumber`.
+  /// Note that this position starts at 0.
+  [propget] HRESULT ColumnNumber([out, retval] UINT32* value);
+
+  /// The Name is the exception's class name.
+  /// In the JSON it is `exceptionDetail.exception.className`.
+  /// This is the empty string if the exception doesn't have a class name.
+  /// This can happen if the script throws a non-Error object such as `throw "abc";`
+  [propget] HRESULT Name([out, retval] LPWSTR* value);
+
+  /// The Message is the exception's message and potentially stack.
+  /// In the JSON it is exceptionDetail.exception.description.
+  /// This is the empty string if the exception doesn't have a description.
+  /// This can happen if the script throws a non-Error object such as throw "abc";.
+  [propget] HRESULT Message([out, retval] LPWSTR* value);
+
+  /// This will return all details of the exception as a JSON string.
+  /// In the case that script has thrown a non-Error object such as `throw "abc";`
+  /// or any other non-Error object, you can get object specific properties.
+  [propget] HRESULT ExceptionAsJSON([out, retval] LPWSTR* value);
+}
+
+/// This is the result for ExecuteScriptWithResult.
+[uuid(D2C59C5C-AD36-4CF4-87CF-2F5359F6D4CB), object, pointer_default(unique)]
+interface ICoreWebView2ExecuteScriptResult : IUnknown {
+
+  /// This property is true if ExecuteScriptWithResult successfully executed script with
```

Comment:
What if the script doesn't have an actual result? Like "just set this value to 6" - there's no result there, really, just a lack of exceptions.